### ISSUE TITLE
Expose basic file I/O

### ISSUE THREAD

**Issue by piodul** [zpp-2022-rust-seastar/seastar-rs#39]
We would like to allow for I/O operations on files.

Proposed interface:

```rust
use std::io;
use std::path::Path;

// A buffer that stores/receives data for I/O operations.
// Its contents are aligned in memory up to 512 bytes.
// read_dma and write_dma require memory to be aligned.
pub struct DmaBuffer {
    chunks: Vec<DmaChunk>,
}

#[repr(align(512))]
struct DmaChunk([u8; 4096]);

// Interface can be modelled after std::fs::OpenOptions
// (a builder)
// https://doc.rust-lang.org/std/fs/struct.OpenOptions.html
pub struct OpenOptions {
    // ...
}

pub struct File {
    // ...
}

// Ref: seastar/core/file.hh
impl File {
    // Read some bytes at given position.
    //
    // The function does not overwrite existing data in the DmaBuffer,
    // but rather reads the data into the free capacity of the internal Vec.
    //
    // Returns the number of bytes read and the original buffer.
    pub async fn read_dma(
        &self,
        buffer: DmaBuffer,
        pos: u64,
    ) -> Result<(usize, DmaBuffer), io::Error>;

    // Writes some bytes at given position.
    //
    // Returns (bytes_written, original_buffer) on success.
    pub async fn write_dma(
        &self,
        buffer: DmaBuffer,
        pos: u64,
    ) -> Result<(usize, DmaBuffer)>;

    pub async fn flush(&self) -> Result<(), io::Error>;
    pub async fn close(&self) -> Result<(), io::Error>;

    pub async fn size(&self) -> Result<u64, io::Error>;
}
```

Caveats:

- It should be safe for all methods of `File` to take `self` by immutable reference.
- Be careful about the buffer lifetime. In Rust, a future can be dropped before the operation it waits on completes. More specifically, it is legal to drop a future which waits until a read or a write finishes - it's important to make sure that the buffer is not dropped until the operation completes. Because of that, the buffers are passed by value (similarly to the design of tokio_uring).
- `io_priority_class` can be skipped for now.
- Nice to have, but not necessary: we could use the `io_intent` to cancel an outstanding read/write operation in case the Rust future is dropped. This does not invalidate the first point, we still must make sure that the buffer lives until the futures returned by `dma_write`/`dma_read` resolve.
